src/core/Akka.Remote/PhiAccrualFailureDetector.cs

```diff
@@ -382,7 +365,8 @@ public double StdDeviation
 
         private static HeartbeatHistory DropOldest(HeartbeatHistory history)
         {
-            return new HeartbeatHistory(history._maxSampleSize, history._intervals.Skip(1).ToList(), history._intervalSum - history._intervals.First(), history._squaredIntervalSum - Pow2(history._intervals.First()));
+            return new HeartbeatHistory(history._maxSampleSize, history._intervals.RemoveAt(0), history._intervalSum - history._intervals.First(), 
```

Comment:
This looks a bit suspect, assuming that code inside the parameters are executed in order from left to right, you removed the first item on the list, and then subtract the sum with the first item on the list. This is, of course, an assumption, but we can't really be sure of the code execution order, that really depends on the compiler and code optimizer.

The sum should be subtracted by the item removed from the list, not the first item on the list, because we don't really know which got executed first.